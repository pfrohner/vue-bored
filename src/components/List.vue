<template>
  <section>
    <table class="table text-left" v-if="$props.activities.length">
      <thead>
        <tr>
          <th scope="col">#</th>
          <th scope="col">Activity</th>
          <th scope="col">Type</th>
          <th scope="col">Participants</th>
        </tr>
      </thead>
      <tbody>
        <tr v-for="activity in $props.activities" :key="activity.key">
          <th scope="row">{{ activity.key }}</th>
          <td>{{ activity.activity }}</td>
          <td>{{ activity.type }}</td>
          <td>{{ activity.participants }}</td>
        </tr>
        <tr>
          <td colspan="4">
            <button
              data-test="clear-btn"
              class="btn btn-danger btn-block"
              @click="$emit('clear')"
            >Clear all</button>
          </td>
        </tr>
      </tbody>
    </table>
    <template v-else>
      <p data-test="empty-msg">Your list is empty.</p>
    </template>
  </section>
</template>

<script>
export default {
  name: "MyList",
  props: {
    activities: {
      type: Array,
      required: true
    }
  }
};
</script>

<style>
.table th,
.table td {
  border-top: 0 !important; /* meh :-/ */
}
</style>
