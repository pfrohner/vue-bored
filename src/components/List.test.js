import { shallowMount } from '@vue/test-utils'
import List from './List.vue'

const setup = () => {
  return shallowMount(List, {
    propsData: {
      activities: [
        {
          key: 1,
          activity: 'test',
          type: 'unit',
          participants: 46
        }
      ]
    }
  })
}

describe('List', () => {
  it('renders correctly', () => {
    const wrapper = setup()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('does not show delete button when list is empty', () => {
    const wrapper = setup()
    wrapper.setProps({
      activities: []
    })
    const button = wrapper.find('[data-test="clear-act-btn"]')

    expect(button.exists()).toBe(false)
  })

  it('shows a placedholder message when list is empty', () => {
    const wrapper = setup()
    wrapper.setProps({
      activities: []
    })
    const msg = wrapper.find('[data-test="empty-msg"]')

    expect(msg.exists()).toBe(true)
  })

  it('emits the correct event when deleting all activities', () => {
    const wrapper = setup()
    const button = wrapper.find('[data-test="clear-btn"]')

    button.trigger('click')

    expect(wrapper.emitted().clear).toBeTruthy()
  })
})
