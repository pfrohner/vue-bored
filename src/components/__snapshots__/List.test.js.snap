// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<section>
  <table
    class="table text-left"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          #
        </th>
         
        <th
          scope="col"
        >
          Activity
        </th>
         
        <th
          scope="col"
        >
          Type
        </th>
         
        <th
          scope="col"
        >
          Participants
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
         
        <td>
          test
        </td>
         
        <td>
          unit
        </td>
         
        <td>
          46
        </td>
      </tr>
       
      <tr>
        <td
          colspan="4"
        >
          <button
            class="btn btn-danger btn-block"
            data-test="clear-btn"
          >
            Clear all
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
